manifests/v3/users.pp

```diff
@@ -0,0 +1,30 @@
+# Create v3 users from user hash
+class simp_snmpd::v3::users(
+  Enum['snmpd','snmptrapd'] $daemon = 'snmpd'
+){
+
+  $simp_snmpd::v3_users_hash.each |String $username,  Hash $settings| {
+    $_authpass = $settings['authpass'] ? {
+      /(undef|UNDEF)/  => passgen("snmp_auth_${username}"),
+      undef            => passgen("snmp_auth_${username}"),
+      default          => $settings['authpass'] }
+    $_privpass = $settings['privpass'] ? {
+      /(undef|UNDEF)/ => passgen("snmp_priv_${username}"),
+      undef           => passgen("snmp_priv_${username}"),
+      default         => $settings['privpass'] }
+    $_authtype =  $settings['authtype'] ? {
+      undef   => $simp_snmpd::defauthtype,
```

Comment:
If we specify a default for this param, it won't be undef unless a user sets it so, is that behavior you intend?